From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: xHyroM <generalkubo@gmail.com>
Date: Sat, 18 Jun 2022 14:29:47 +0200
Subject: [PATCH] Add ability to toggle default commands


diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
index 9b9af189b3865055570ba533b97c92c105cc5a37..d5a598f13ee9b793022a7914c17d1aad790990f7 100644
--- a/src/main/java/org/bukkit/Bukkit.java
+++ b/src/main/java/org/bukkit/Bukkit.java
@@ -2307,6 +2307,44 @@ public final class Bukkit {
         return server.permissionMessage();
     }
 
+    // Sharkur start - Add ability to toggle default commands
+    /**
+     * Checks if the version command is enabled
+     *
+     * @return boolean
+     */
+    public static boolean isVersionCommandEnabled() {
+        return server.isVersionCommandEnabled();
+    }
+
+    /**
+     * Checks if the plugins command is enabled
+     *
+     * @return boolean
+     */
+    public static boolean isPluginsCommandEnabled() {
+        return server.isPluginsCommandEnabled();
+    }
+
+    /**
+     * Checks if the reload command is enabled
+     *
+     * @return boolean
+     */
+    public static boolean isReloadCommandEnabled() {
+        return server.isReloadCommandEnabled();
+    }
+
+    /**
+     * Checks if the help command is enabled
+     *
+     * @return boolean
+     */
+    public static boolean isHelpCommandEnabled() {
+        return server.isHelpCommandEnabled();
+    }
+    // Sharkur end
+
     /**
      * Creates a PlayerProfile for the specified uuid, with name as null.
      *
diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index 35df0d2bd3ebe0feadb8ede8383acbd9938fa9f1..62aa5fad8cc11dbbc07838d815bb8f95bc98c2dc 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -2020,6 +2020,36 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
      */
     @NotNull net.kyori.adventure.text.Component permissionMessage();
 
+    // Sharkur start - Add ability to toggle default commands
+    /**
+     * Checks if the version command is enabled
+     *
+     * @return boolean
+     */
+    boolean isVersionCommandEnabled();
+
+    /**
+     * Checks if the plugins command is enabled
+     *
+     * @return boolean
+     */
+    boolean isPluginsCommandEnabled();
+
+    /**
+     * Checks if the reload command is enabled
+     *
+     * @return boolean
+     */
+    boolean isReloadCommandEnabled();
+
+    /**
+     * Checks if the help command is enabled
+     *
+     * @return boolean
+     */
+    boolean isHelpCommandEnabled();
+    // Sharkur end
+
     /**
      * Creates a PlayerProfile for the specified uuid, with name as null.
      *
diff --git a/src/main/java/org/bukkit/command/SimpleCommandMap.java b/src/main/java/org/bukkit/command/SimpleCommandMap.java
index 0bc24d0effe9b2e44c41a1c00060b0ebf7395c4a..6e6a991d46ece6b559f56b469199be425de2c7b5 100644
--- a/src/main/java/org/bukkit/command/SimpleCommandMap.java
+++ b/src/main/java/org/bukkit/command/SimpleCommandMap.java
@@ -27,18 +27,20 @@ public class SimpleCommandMap implements CommandMap {
 
     public SimpleCommandMap(@NotNull final Server server) {
         this.server = server;
-        setDefaultCommands();
+        // setDefaultCommands(); // Sharkur - Move to SharkurConfig
     }
 
-    private void setDefaultCommands() {
-        register("bukkit", new VersionCommand("version"));
-        register("bukkit", new ReloadCommand("reload"));
-        register("bukkit", new PluginsCommand("plugins"));
+    public void setDefaultCommands() { // Sharkur - private -> public
+        // Sharkur start - Add ability to toggle default commands
+        if (server.isVersionCommandEnabled()) register("bukkit", new VersionCommand("version"));
+        if (server.isReloadCommandEnabled()) register("bukkit", new ReloadCommand("reload"));
+        if (server.isPluginsCommandEnabled()) register("bukkit", new PluginsCommand("plugins"));
+        // Sharkur end
         register("bukkit", new co.aikar.timings.TimingsCommand("timings")); // Paper
     }
 
     public void setFallbackCommands() {
-        register("bukkit", new HelpCommand());
+        if (server.isHelpCommandEnabled()) register("bukkit", new HelpCommand()); // Sharkur - Add ability to toggle default commands
     }
 
     /**
