From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: xHyroM <generalkubo@gmail.com>
Date: Sat, 18 Jun 2022 14:29:47 +0200
Subject: [PATCH] Add ability to toggle default commands


diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
index 9222adc75b68b9a899b4eca2a1803844b9cdbe26..1008f37ab20a52a397ca5e192af81ef768af5ad6 100644
--- a/src/main/java/org/bukkit/Bukkit.java
+++ b/src/main/java/org/bukkit/Bukkit.java
@@ -2290,6 +2290,44 @@ public final class Bukkit {
         return server.permissionMessage();
     }
 
+    // Sharkur start - Add ability to toggle default commands
+    /**
+     * Checks if the version command is enabled
+     *
+     * @return boolean
+     */
+    public static boolean isVersionCommandEnabled() {
+        return server.isVersionCommandEnabled();
+    }
+
+    /**
+     * Checks if the plugins command is enabled
+     *
+     * @return boolean
+     */
+    public static boolean isPluginsCommandEnabled() {
+        return server.isPluginsCommandEnabled();
+    }
+
+    /**
+     * Checks if the reload command is enabled
+     *
+     * @return boolean
+     */
+    public static boolean isReloadCommandEnabled() {
+        return server.isReloadCommandEnabled();
+    }
+
+    /**
+     * Checks if the help command is enabled
+     *
+     * @return boolean
+     */
+    public static boolean isHelpCommandEnabled() {
+        return server.isHelpCommandEnabled();
+    }
+    // Sharkur end
+
     /**
      * Creates a PlayerProfile for the specified uuid, with name as null.
      *
diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index af6c76ff488af2cd4de1efa89b7983720aa14fcd..1cc098ebb99c92fa2aac6c81042c4a6f35d1cd54 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -2005,6 +2005,36 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
      */
     @NotNull net.kyori.adventure.text.Component permissionMessage();
 
+    // Sharkur start - Add ability to toggle default commands
+    /**
+     * Checks if the version command is enabled
+     *
+     * @return boolean
+     */
+    boolean isVersionCommandEnabled();
+
+    /**
+     * Checks if the plugins command is enabled
+     *
+     * @return boolean
+     */
+    boolean isPluginsCommandEnabled();
+
+    /**
+     * Checks if the reload command is enabled
+     *
+     * @return boolean
+     */
+    boolean isReloadCommandEnabled();
+
+    /**
+     * Checks if the help command is enabled
+     *
+     * @return boolean
+     */
+    boolean isHelpCommandEnabled();
+    // Sharkur end
+
     /**
      * Creates a PlayerProfile for the specified uuid, with name as null.
      *
diff --git a/src/main/java/org/bukkit/command/SimpleCommandMap.java b/src/main/java/org/bukkit/command/SimpleCommandMap.java
index 0bc24d0effe9b2e44c41a1c00060b0ebf7395c4a..6e6a991d46ece6b559f56b469199be425de2c7b5 100644
--- a/src/main/java/org/bukkit/command/SimpleCommandMap.java
+++ b/src/main/java/org/bukkit/command/SimpleCommandMap.java
@@ -27,18 +27,20 @@ public class SimpleCommandMap implements CommandMap {
 
     public SimpleCommandMap(@NotNull final Server server) {
         this.server = server;
-        setDefaultCommands();
+        // setDefaultCommands(); // Sharkur - Move to SharkurConfig
     }
 
-    private void setDefaultCommands() {
-        register("bukkit", new VersionCommand("version"));
-        register("bukkit", new ReloadCommand("reload"));
-        register("bukkit", new PluginsCommand("plugins"));
+    public void setDefaultCommands() { // Sharkur - private -> public
+        // Sharkur start - Add ability to toggle default commands
+        if (server.isVersionCommandEnabled()) register("bukkit", new VersionCommand("version"));
+        if (server.isReloadCommandEnabled()) register("bukkit", new ReloadCommand("reload"));
+        if (server.isPluginsCommandEnabled()) register("bukkit", new PluginsCommand("plugins"));
+        // Sharkur end
         register("bukkit", new co.aikar.timings.TimingsCommand("timings")); // Paper
     }
 
     public void setFallbackCommands() {
-        register("bukkit", new HelpCommand());
+        if (server.isHelpCommandEnabled()) register("bukkit", new HelpCommand()); // Sharkur - Add ability to toggle default commands
     }
 
     /**
