From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: xHyroM <generalkubo@gmail.com>
Date: Sun, 3 Jul 2022 19:21:55 +0200
Subject: [PATCH] Add publishing to reposilite


diff --git a/build.gradle.kts b/build.gradle.kts
index a141e5cdcec4d09b05252782e8a0ab5fac532680..ed17acda6e18203cad7a8ca30e3d0191ddae5a81 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -153,3 +153,18 @@ tasks.check {
     dependsOn(scanJar)
 }
 // Paper end
+
+// Sharkur start
+publishing {
+    repositories {
+        maven {
+            name = "sharkur"
+            url = uri("https://reposilite.cezarsalat.tk/releases")
+            credentials(PasswordCredentials::class)
+            authentication {
+                create<BasicAuthentication>("basic")
+            }
+        }
+    }
+}
+// Sharkur end
\ No newline at end of file
