From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: xHyroM <generalkubo@gmail.com>
Date: Mon, 4 Jul 2022 19:20:25 +0200
Subject: [PATCH] Include BungeeGuard


diff --git a/src/main/java/org/bukkit/plugin/SimplePluginManager.java b/src/main/java/org/bukkit/plugin/SimplePluginManager.java
index cc67217ff6003c83f10c6eeb7d4a2205cb0129c4..780ff817833f77275e1d89c3cf9775b6a3c4d9ce 100644
--- a/src/main/java/org/bukkit/plugin/SimplePluginManager.java
+++ b/src/main/java/org/bukkit/plugin/SimplePluginManager.java
@@ -171,6 +171,12 @@ public final class SimplePluginManager implements PluginManager {
                 continue;
             }
             // Purpur end
+            // Sharkur start
+            if (plugins.containsKey(description.getName()) && description.getName().equalsIgnoreCase("bungeeguard")) {
+                server.getLogger().log(Level.INFO, "Sharkur: Using user-provided BungeeGuard plugin instead of our own.");
+                continue;
+            }
+            // Sharkur end
 
             File replacedFile = plugins.put(description.getName(), file);
             if (replacedFile != null) {
