From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: xHyroM <generalkubo@gmail.com>
Date: Sat, 18 Jun 2022 13:12:56 +0200
Subject: [PATCH] Add ability to toggle default commands


diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
index 44673fcec96a075bd02800b51318832d82b63c7e..f75b21ffca1cad5b82843ef224570f9f56ab0cb1 100644
--- a/src/main/java/org/bukkit/Bukkit.java
+++ b/src/main/java/org/bukkit/Bukkit.java
@@ -2259,6 +2259,35 @@ public final class Bukkit {
         return server.permissionMessage();
     }
 
+    // Sharkur start - Add ability to toggle default commands
+    /**
+     * Checks if the version command is enabled
+     *
+     * @return boolean
+     */
+    public boolean isVersionCommandEnabled() {
+        return server.isVersionCommandEnabled();
+    }
+
+    /**
+     * Checks if the plugins command is enabled
+     *
+     * @return boolean
+     */
+    public boolean isPluginsCommandEnabled() {
+        return server.isPluginsCommandEnabled();
+    }
+
+    /**
+     * Checks if the reload command is enabled
+     *
+     * @return boolean
+     */
+    public boolean isReloadCommandEnabled() {
+        return server.isReloadCommandEnabled();
+    }
+    // Sharkur end
+
     /**
      * Creates a PlayerProfile for the specified uuid, with name as null.
      *
diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index 1be0365d32aaca0098a05b6c667ced33a4f20f7e..cf1cd77bb4aba4af690926a817aeaf026bfb823b 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -1980,6 +1980,27 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
      */
     @NotNull net.kyori.adventure.text.Component permissionMessage();
 
+    // Sharkur start - Add ability to toggle default commands
+    /**
+     * Checks if the version command is enabled
+     *
+     * @return boolean
+     */
+    boolean isVersionCommandEnabled();
+    /**
+     * Checks if the plugins command is enabled
+     *
+     * @return boolean
+     */
+    boolean isPluginsCommandEnabled();
+    /**
+     * Checks if the reload command is enabled
+     *
+     * @return boolean
+     */
+    boolean isReloadCommandEnabled();
+    // Sharkur end
+
     /**
      * Creates a PlayerProfile for the specified uuid, with name as null.
      *
diff --git a/src/main/java/org/bukkit/command/SimpleCommandMap.java b/src/main/java/org/bukkit/command/SimpleCommandMap.java
index 0bc24d0effe9b2e44c41a1c00060b0ebf7395c4a..c4234b92c96c6a8ab01df2c05309ea60b7b19b04 100644
--- a/src/main/java/org/bukkit/command/SimpleCommandMap.java
+++ b/src/main/java/org/bukkit/command/SimpleCommandMap.java
@@ -31,9 +31,11 @@ public class SimpleCommandMap implements CommandMap {
     }
 
     private void setDefaultCommands() {
-        register("bukkit", new VersionCommand("version"));
-        register("bukkit", new ReloadCommand("reload"));
-        register("bukkit", new PluginsCommand("plugins"));
+        // Sharkur start - Add ability to toggle default commands
+        if (server.isVersionCommandEnabled()) register("bukkit", new VersionCommand("version"));
+        if (server.isReloadCommandEnabled()) register("bukkit", new ReloadCommand("reload"));
+        if (server.isPluginsCommandEnabled()) register("bukkit", new PluginsCommand("plugins"));
+        // Sharkur end
         register("bukkit", new co.aikar.timings.TimingsCommand("timings")); // Paper
     }
 
