From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: xHyroM <generalkubo@gmail.com>
Date: Sat, 18 Jun 2022 14:29:40 +0200
Subject: [PATCH] Add ability to toggle default commands


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index a3549341d6107ed5cf1c22c610322eff8359a4db..881550dc93e62f6f76f79d107d63019e7e6146c5 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -2950,6 +2950,28 @@ public final class CraftServer implements Server {
         return io.papermc.paper.configuration.GlobalConfiguration.get().messages.noPermission;
     }
 
+    // Sharkur start - Add ability to toggle default commands
+    @Override
+    public boolean isVersionCommandEnabled() {
+        return org.sharkurmc.sharkur.SharkurConfig.enableVersionCommand;
+    }
+
+    @Override
+    public boolean isPluginsCommandEnabled() {
+        return org.sharkurmc.sharkur.SharkurConfig.enablePluginsCommand;
+    }
+
+    @Override
+    public boolean isReloadCommandEnabled() {
+        return org.sharkurmc.sharkur.SharkurConfig.enableReloadCommand;
+    }
+
+    @Override
+    public boolean isHelpCommandEnabled() {
+        return org.sharkurmc.sharkur.SharkurConfig.enableHelpCommand;
+    }
+    // Sharkur end
+
     @Override
     public com.destroystokyo.paper.profile.PlayerProfile createProfile(@Nonnull UUID uuid) {
         return createProfile(uuid, null);
diff --git a/src/main/java/org/sharkurmc/sharkur/SharkurConfig.java b/src/main/java/org/sharkurmc/sharkur/SharkurConfig.java
index c01730304d5b1bf32bc73b36aaf982480e3106c0..40e9cbeca8f76e8d49ffab2c8894c21c59fcab0f 100644
--- a/src/main/java/org/sharkurmc/sharkur/SharkurConfig.java
+++ b/src/main/java/org/sharkurmc/sharkur/SharkurConfig.java
@@ -154,4 +154,17 @@ public class SharkurConfig {
     private static void messages() {
         sharkurCommandsPrefix = getString("settings.messages.sharkur-commands-prefix", sharkurCommandsPrefix);
     }
+
+    public static boolean enableVersionCommand = true;
+    public static boolean enablePluginsCommand = true;
+    public static boolean enableReloadCommand = true;
+    public static boolean enableHelpCommand = true;
+    private static void commands() {
+        enableVersionCommand = getBoolean("settings.commands.version", enableVersionCommand);
+        enablePluginsCommand = getBoolean("settings.commands.plugins", enablePluginsCommand);
+        enableReloadCommand = getBoolean("settings.commands.reload", enableReloadCommand);
+        enableHelpCommand = getBoolean("settings.commands.help", enableHelpCommand);
+
+        MinecraftServer.getServer().server.getCommandMap().setDefaultCommands();
+    }
 }
