From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: xHyroM <generalkubo@gmail.com>
Date: Sun, 19 Jun 2022 08:43:24 +0200
Subject: [PATCH] Add PlayerClientBrand event


diff --git a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
index d09c8aab9112e2949dbb9d16a135fa2122c1db2f..5be3365724eb935ab379e3d10e9812316867c0d6 100644
--- a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -3747,6 +3747,7 @@ public class ServerGamePacketListenerImpl implements ServerPlayerConnection, Tic
                     } catch (StringIndexOutOfBoundsException ex) {
                         this.clientBrandName = "illegal";
                     }
+                    this.cserver.getPluginManager().callEvent(new org.sharkurmc.sharkur.event.PlayerClientBrandEvent(player.getBukkitEntity(), clientBrandName)); // Sharkur - Add PlayerClientBrand event
                 }
                 // Paper end
                 this.cserver.getMessenger().dispatchIncomingMessage(this.player.getBukkitEntity(), packet.identifier.toString(), data);
