From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: xHyroM <generalkubo@gmail.com>
Date: Sun, 19 Jun 2022 08:43:24 +0200
Subject: [PATCH] Add PlayerClientBrand event


diff --git a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
index 7f2dc4f85154a27e547aa1860eca1c0bfbb3e6ee..1f07a8838aaee2511a8203cebf83018b3c945229 100644
--- a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -3534,6 +3534,7 @@ public class ServerGamePacketListenerImpl implements ServerPlayerConnection, Ser
                     } catch (StringIndexOutOfBoundsException ex) {
                         this.clientBrandName = "illegal";
                     }
+                    this.cserver.getPluginManager().callEvent(new org.sharkurmc.sharkur.event.PlayerClientBrandEvent(player.getBukkitEntity(), clientBrandName)); // Sharkur - Add PlayerClientBrand event
                 }
                 // Paper end
                 this.cserver.getMessenger().dispatchIncomingMessage(this.player.getBukkitEntity(), packet.identifier.toString(), data);
