From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: xHyroM <generalkubo@gmail.com>
Date: Sun, 19 Jun 2022 08:43:24 +0200
Subject: [PATCH] Add PlayerClientBrand event


diff --git a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
index 322bb76994dc3bf3eb21ac6c41c6d744743c766f..2fb2257ab3a7e2603b71ff11d097a68be3fc7a6f 100644
--- a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -3745,6 +3745,7 @@ public class ServerGamePacketListenerImpl implements ServerPlayerConnection, Tic
                     } catch (StringIndexOutOfBoundsException ex) {
                         this.clientBrandName = "illegal";
                     }
+                    this.cserver.getPluginManager().callEvent(new org.sharkurmc.sharkur.event.PlayerClientBrandEvent(player.getBukkitEntity(), clientBrandName)); // Sharkur - Add PlayerClientBrand event
                 }
                 // Paper end
                 this.cserver.getMessenger().dispatchIncomingMessage(this.player.getBukkitEntity(), packet.identifier.toString(), data);
