From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: xHyroM <generalkubo@gmail.com>
Date: Sun, 19 Jun 2022 08:43:24 +0200
Subject: [PATCH] Add PlayerClientBrand event


diff --git a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
index 38cd5aec3c54984c3c3429aa0bfb525ba170f116..c044ead6ba19dc451be9730329be0b7370f965fe 100644
--- a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -3525,6 +3525,7 @@ public class ServerGamePacketListenerImpl implements ServerPlayerConnection, Ser
                     } catch (StringIndexOutOfBoundsException ex) {
                         this.clientBrandName = "illegal";
                     }
+                    this.cserver.getPluginManager().callEvent(new org.sharkurmc.sharkur.event.PlayerClientBrandEvent(player.getBukkitEntity(), clientBrandName)); // Sharkur - Add PlayerClientBrand event
                 }
                 // Paper end
                 this.cserver.getMessenger().dispatchIncomingMessage(this.player.getBukkitEntity(), packet.identifier.toString(), data);
