From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: xHyroM <generalkubo@gmail.com>
Date: Mon, 20 Jun 2022 19:37:22 +0200
Subject: [PATCH] Add ability to hide player's ip in logs


diff --git a/src/main/java/net/minecraft/server/players/PlayerList.java b/src/main/java/net/minecraft/server/players/PlayerList.java
index fecd65dc87a99a5692c59dd590b20327b1eb31c2..a24541fffae4df499d90c1436f90d083665a6bb8 100644
--- a/src/main/java/net/minecraft/server/players/PlayerList.java
+++ b/src/main/java/net/minecraft/server/players/PlayerList.java
@@ -498,7 +498,13 @@ public abstract class PlayerList {
         // Paper end
         org.purpurmc.purpur.task.BossBarTask.addToAll(player); // Purpur
         // CraftBukkit - Moved from above, added world
-        PlayerList.LOGGER.info("{}[{}] logged in with entity id {} at ([{}]{}, {}, {})", player.getName().getString(), s1, player.getId(), worldserver1.serverLevelData.getLevelName(), player.getX(), player.getY(), player.getZ());
+        // Sharkur - Add ability to hide player's ip in logs
+        if (org.sharkurmc.sharkur.SharkurConfig.hidePlayersIp) {
+            PlayerList.LOGGER.info("{}[/x.x.x.x:xxxxx] logged in with entity id {} at ([{}]{}, {}, {})", player.getName().getString(), player.getId(), worldserver1.serverLevelData.getLevelName(), player.getX(), player.getY(), player.getZ());
+        } else {
+            PlayerList.LOGGER.info("{}[{}] logged in with entity id {} at ([{}]{}, {}, {})", player.getName().getString(), s1, player.getId(), worldserver1.serverLevelData.getLevelName(), player.getX(), player.getY(), player.getZ());
+        }
+        // Sharkur end
     }
 
     public void updateEntireScoreboard(ServerScoreboard scoreboard, ServerPlayer player) {
diff --git a/src/main/java/org/sharkurmc/sharkur/SharkurConfig.java b/src/main/java/org/sharkurmc/sharkur/SharkurConfig.java
index 89fa483516e5643e80207cbd4e100deebbe59d16..110762fd7e47182bb9c28a9649ea282ce22504dc 100644
--- a/src/main/java/org/sharkurmc/sharkur/SharkurConfig.java
+++ b/src/main/java/org/sharkurmc/sharkur/SharkurConfig.java
@@ -172,4 +172,9 @@ public class SharkurConfig {
     private static void disableUserCache() {
         disableUserCache = getBoolean("settings.disable-user-cache", disableUserCache);
     }
+
+    public static boolean hidePlayersIp = false;
+    private static void hidePlayersIp() {
+        hidePlayersIp = getBoolean("settings.hide-players-ip", hidePlayersIp);
+    }
 }
